docs/pnpm.md

```diff
@@ -0,0 +1,282 @@
+# pnpm and rules_js
+
+rules_js models dependency handling on [pnpm](https://pnpm.io/). Our design goal is to closely mimic pnpm's behavior.
+
+To start, some non-Bazel-specific tool (typically pnpm) which performs dependency resolutions
+and solves version constraints. It also determines how the `node_modules` tree will be structured for runtime.
+This information is encoded into a lockfile which is checked into the source repository.
+
+The pnpm lockfile format includes all the information needed to define [`npm_import`](/docs/npm_import.md#npm_import) rules,
+allowing Bazel's downloader to do the fetches. This info includes the integrity hash, as calculated by the package manager,
+so that Bazel can guarantee supply-chain security.
+
+For more details, see <https://github.com/pnpm/pnpm/blob/main/packages/lockfile-types/src/index.ts>.
+
+Bazel will only fetch the packages which are required for the requested targets to be analyzed.
+Thus it is performant to convert a very large `pnpm-lock.yaml` file without concern for
+users needing to fetch many unnecessary packages. We have benchmarked this code with
+800+ importers and ~15,000 npm packages to run in 3sec, when Bazel determines that an input changed.
+
+While the [`npm_import`](/docs/npm_import.md#npm_import) rule can be used to bring individual packages into Bazel,
+most users will want to import their entire lockfile.
+The `npm_translate_lock` rule does this, and its operation is described below.
+You may wish to read the [generated API documentation](/docs/npm_import.md#npm_translate_lock) as well.
+
+## Using npm_translate_lock
+
+In `WORKSPACE`, call the repository rule pointing to your `pnpm-lock.yaml` file:
+
+```starlark
+load("@aspect_rules_js//npm:npm_import.bzl", "npm_translate_lock")
+
+# Read the pnpm-lock.yaml file to automate creation of npm_import rules
+npm_translate_lock(
+    # Creates a new repository named "@npm_deps"
+    name = "npm_deps",
```

Comment:
switching out our `@npm` convention?

Reply:
should probably match what the release snippet uses here

Reply:
this was intentionally using a different name to show that `npm` isn't magic... but yeah we wouldn't want most users to do this. I'll change it and mention in the text